### ISSUE TITLE
Add of ignored files from status window leaves the list out of sync

### ISSUE THREAD

**Issue by tipa40** [tpope/vim-fugitive#2099]
Hi,
Thank you for a great tool! I know this is a bit of a special case but... When I add files that are on the ignore list using - or s from the :Git view having my "git config advice.addIgnoredFile false"(regardless of being false/true actually) I get the error message telling me about that the file is on my ignore list and the file is still on the unstaged list even though it is actually staged, as can be seen issuing the :Git once more.

I would think it would be great if (I would still be allowed to add ignored files, at least when having my config flag set to false, and the warning/error could also be there, and that) the status list would be updated to reflect the change with no need to manually further run :Git.

**Comment by tpope** [comment]
I don't understand. Ignored files shouldn't be showing up in the `:Git` window to begin with.

**Comment by tipa40** [comment]
Yes you are right, maybe we should close this. 
The reason some of them are shown is that I have committed some of them earlier and those are shown, the others not.
I should probably rethink my workflow and use the .gitignore the way it is intended.

**Comment by tpope** [comment]
If there is a reload issue, even with your unconventional practices, I am open to fixing that. But I leave it to you to figure out why your unconventional practices cause the issue.

**Comment by tipa40** [comment]
Ok sounds great. :) My workflow is out of laziness: I have my ~/.config directory which I want selectively version controlled in git, but I wanted to not have to keep maintaining my .gitigore adding the directories which I am not interested in, and I don't like to see a growing list of files that I am not interested in, in my status list, and at the same time I want to use some tool like vim-fugitive or lazygit. (The "git status -uno" is available at the command line.)

So, at the time of making this workflow decision I found an ugly workaround in that I in my .gitignore instead could invert it to ignore all files(, using just one wildcard line), and then adding the ones I am interested in(, with "git add -f" from command line), since it's less work, whenever a new one that I'd actually like to version control appears, instead of the other way. This because an ignored file shows up in the status list if it is in version control. But I will of course need to take care not to miss any new file that I would like to add.

So I tried fugitive status view using hyphen to add one of my ignored files that is already in my repository and the view remained unchanged although I could see using :G one more time to update that it jumped to the "Staged" section meaning my previous action actually was successful, just that it didn't show.
